### PR TITLE
Feature/mail

### PR DESCRIPTION
**Pull request by susithrupasinghe**
## Purpose

Sending Mails to user/ set of users #6
## Goals

1. Sending emails
2. Checking availability of SMTP

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have performed a self-review of my own code
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

File: src/api/mails/mail.constants.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+
```
Comment (ebonynon) [inline src/api/mails/mail.constants.js:2]: Remove empty two lines

File: src/api/mails/mail.controller.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  });
+
+
+const sendMail = async(req, res, next) => {
+
+    logger.info('mail.controller.js sendMail(): ' + req.body);
+
+    try {
+      
+
```
Comment (ebonynon) [inline src/api/mails/mail.controller.js:33]: Remove unnecessary empty lines on code blocks

File: src/api/routes.js [hunk 3]
```diff
@@ -2,11 +2,13 @@ import { Router } from 'express';
...
 import authRouter from './auth/routes';
 import { handleHealthRequest } from './health';
+import  mailRouter  from './mails/routes';
 
 const router = Router();
 
 router.get('/health', handleHealthRequest);
 router.use('/events', eventsRouter);
 router.use('/', authRouter);
+router.use('/api/v1/mail', mailRouter);
```
Comment (ebonynon) [inline src/api/routes.js:12]: My bad bn, can you change as `/mail`

**Review by ebonynon — COMMENTED** [review]
There is some code formatting issues. Remove unnecessary line gaps and try [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode#:~:text=Prettier%20Formatter%20for%20Visual%20Studio%20Code%20Prettier%20is,%C2%B7%20TypeScript%20%C2%B7%20Flow%20%C2%B7%20JSX%20%C2%B7%20JSON)

File: src/api/routes.js (hunk 3 above)
Comment (susithrupasinghe) [reply src/api/routes.js]: Fixed

File: src/api/mails/mail.controller.js (hunk 2 above)
Comment (susithrupasinghe) [reply src/api/mails/mail.controller.js]: Fixed

File: src/api/mails/mail.constants.js (hunk 1 above)
Comment (susithrupasinghe) [reply src/api/mails/mail.constants.js]: fixed

File: src/api/mails/mail.constants.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const CREDENTIALS = {
+  HOST: "smtp.mailtrap.io",
+  PORT: "2525",
+  USER: "c8cca3941aea9c",
+  PASSWORD: "18763bafd470a9",
+};
+
+export default {
+  CREDENTIALS,
+};
```
Comment (Shehanka) [inline src/api/mails/mail.constants.js:10]: Don't push secrets to public repository. Gitignore it. when we're deploying the app we can put this as a config file. But still wondering how we configure this for github actions 🤨. Need to find a way

**Review by Shehanka — CHANGES_REQUESTED** [review]
PR is overall good. But need to find a way for that. 

@ebonynon @susithrupasinghe Let's discuss about that in discussion

**Review by Shehanka — APPROVED** [review]

**Comment by Shehanka** [comment]
Gonna merge this for now. Let's discuss the issue in here https://github.com/sliit-foss/rsvp/discussions/29

**Pull request merged**